embedded/sql/stmt.go

```diff
@@ -2185,17 +2175,120 @@ func (stmt *SelectStmt) genScanSpecs(tx *SQLTx, params map[string]interface{}) (
 	}
 
 	return &ScanSpecs{
-		index:         sortingIndex,
+		Index:         sortingIndex,
 		rangesByColID: rangesByColID,
-		descOrder:     descOrder,
+		DescOrder:     descOrder,
 	}, nil
 }
 
 type tableRef struct {
-	db       string
-	table    string
-	asBefore uint64
-	as       string
+	db     string
+	table  string
+	period period
+	as     string
+}
+
+type period struct {
+	start *openPeriod
+	end   *openPeriod
+}
+
+type openPeriod struct {
+	inclusive bool
+	instant   periodInstant
+}
+
+type periodInstant struct {
+	exp         ValueExp
+	instantType instantType
+}
+
+type instantType = int
+
+const (
+	txInstant instantType = iota
+	timeInstant
+)
+
+func (i periodInstant) resolve(tx *SQLTx, params map[string]interface{}, asc, inclusive bool) (uint64, error) {
+	exp, err := i.exp.substitute(params)
+	if err != nil {
+		return 0, err
+	}
+
+	instantVal, err := exp.reduce(tx.catalog, nil, tx.currentDB.name, "")
+	if err != nil {
+		return 0, err
+	}
+
+	if i.instantType == txInstant {
+		txID, ok := instantVal.Value().(int64)
+		if !ok {
+			return 0, fmt.Errorf("%w: invalid tx range", ErrIllegalArguments)
```

Comment:
That would be clearly due to an invalid type used, maybe let's have some more descriptive error message here:

```suggestion
			return 0, fmt.Errorf("%w: invalid tx range, tx ID must be an integer, %s given", ErrIllegalArguments, instantVal.Type())
```